oap-server/server-bootstrap/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java

```diff
@@ -105,6 +109,26 @@ private void overrideConfigBySystemEnv(ApplicationConfiguration configuration) {
         }
     }
 
+    private void selectConfig(final Map<String, Map<String, Object>> moduleConfiguration) {
+        final Set<String> modulesWithoutProvider = new HashSet<>();
+        for (final Map.Entry<String, Map<String, Object>> entry : moduleConfiguration.entrySet()) {
+            final String moduleName = entry.getKey();
+            final Map<String, Object> providerConfig = entry.getValue();
+            if (providerConfig.containsKey("selector")) {
+                final String selector = (String) providerConfig.get("selector");
+                final String resolvedSelector = PropertyPlaceholderHelper.INSTANCE.replacePlaceholders(
+                    selector, System.getProperties()
+                );
+                providerConfig.entrySet().removeIf(e -> !resolvedSelector.equals(e.getKey()));
```

Comment:
@kezhenxu94 To be more clear,  after this remove, it could be no provider available, we should check and report error immidiately.